kiwi/system/uri.py

```diff
@@ -91,38 +92,41 @@ def translate(self):
             raise KiwiUriStyleUnknown(
                 'URI scheme not detected {uri}'.format(uri=self.uri)
             )
-
-        if uri.scheme == 'obs' and self.repo_type == 'yast2':
-            return self._obs_distribution(
-                ''.join([uri.netloc, uri.path])
-            )
         elif uri.scheme == 'obs':
-            return self._obs_project(
-                ''.join([uri.netloc, uri.path])
-            )
+            if check_build_environment and Defaults.is_buildservice_worker():
+                return self._buildservice_path(
+                    name=''.join([uri.netloc, uri.path]),
+                    fragment=uri.fragment,
+                    urischeme=uri.scheme
+                )
+            else:
+                if self.repo_type == 'yast2':
+                    return self._obs_distribution(
+                        ''.join([uri.netloc, uri.path])
+                    )
+                else:
+                    return self._obs_project(
+                        ''.join([uri.netloc, uri.path])
+                    )
         elif uri.scheme == 'obsrepositories':
-            return self._suse_buildservice_path(
+            if not Defaults.is_buildservice_worker():
+                raise KiwiUriStyleUnknown(
+                    'Only the buildservice can use the {0} schema'.format(
+                        uri.scheme
+                    )
+                )
+            return self._buildservice_path(
                 name=''.join([uri.netloc, uri.path]),
                 fragment=uri.fragment,
                 urischeme=uri.scheme
             )
-        elif uri.scheme == 'ibs':
-            return self._ibs_project(
-                ''.join([uri.netloc, uri.path])
-            )
         elif uri.scheme == 'dir':
             return self._local_path(uri.path)
         elif uri.scheme == 'file':
             return self._local_path(uri.path)
         elif uri.scheme == 'iso':
             return self._iso_mount_path(uri.path)
-        elif uri.scheme == 'suse':
-            return self._suse_buildservice_path(
-                name=''.join([uri.netloc, uri.path]),
-                fragment=uri.fragment,
-                urischeme=uri.scheme
-            )
-        elif uri.scheme == 'http' or uri.scheme == 'https' or uri.scheme == 'ftp':
+        elif uri.scheme.startswith('http') or uri.scheme == 'ftp':
```

Comment:
starts with http: covers http and https, all tests are still passing, well that would also cover httpx :)